AlterTabletIT is broken

On CICD it shows zero tests and failure:
```
[INFO] Running com.scylladb.cdc.lib.AlterTableIT
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
Oct 16, 2025 6:11:42 PM com.scylladb.cdc.model.master.Master runUntilException
INFO: Master found a new generation: GenerationId(Timestamp(16/10/2025, 18:11:40.189)). Will call transport.configureWorkers().
Error: Exception in thread "Thread-1" java.lang.RuntimeException: java.lang.InterruptedException: sleep interrupted
	at com.scylladb.cdc.lib.AlterTableIT.lambda$alterBaseTableAtRuntime$0(AlterTableIT.java:69)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.InterruptedException: sleep interrupted
	at java.base/java.lang.Thread.sleep(Native Method)
	at com.scylladb.cdc.lib.AlterTableIT.lambda$alterBaseTableAtRuntime$0(AlterTableIT.java:67)
	... 1 more
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 60.46 s -- in com.scylladb.cdc.lib.AlterTableIT
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] 
[INFO] --- docker:0.46.0:stop (stop) @ scylla-cdc-lib ---
[INFO] DOCKER> [scylladb/scylla:latest] "scylla": Stop and removed container 7d6329bef219 after 0 ms
[INFO] 
[INFO] --- failsafe:3.5.4:verify (default) @ scylla-cdc-lib ---
[INFO] 
[INFO] -----------------< com.scylladb:scylla-cdc-replicator >-----------------
[INFO] Building Java Library for Scylla CDC - replicator 1.3.8-SNAPSHOT   [5/6]
[INFO]   from scylla-cdc-replicator/pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-assembly-plugin/3.7.1/maven-assembly-plugin-3.7.1.pom
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-assembly-plugin/3.7.1/maven-assembly-plugin-3.7.1.pom (15 kB at 324 kB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-assembly-plugin/3.7.1/maven-assembly-plugin-3.7.1.jar
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-assembly-plugin/3.7.1/maven-assembly-plugin-3.7.1.jar (240 kB at 1.4 MB/s)
[INFO] 
[INFO] --- resources:3.3.1:resources (default-resources) @ scylla-cdc-replicator ---
[INFO] Copying 1 resource from src/main/resources to target/classes
[INFO] 
[INFO] --- compiler:3.13.0:compile (default-compile) @ scylla-cdc-replicator ---
[INFO] Recompiling the module because of changed dependency.
[INFO] Compiling 24 source files with javac [debug target 11] to target/classes
Warning:  /home/runner/work/scylla-cdc-java/scylla-cdc-java/scylla-cdc-replicator/src/main/java/com/scylladb/cdc/replicator/Main.java:[16,16] sun.misc.Signal is internal proprietary API and may be removed in a future release
Warning:  /home/runner/work/scylla-cdc-java/scylla-cdc-java/scylla-cdc-replicator/src/main/java/com/scylladb/cdc/replicator/Main.java:[78,35] sun.misc.Signal is internal proprietary API and may be removed in a future release
Warning:  /home/runner/work/scylla-cdc-java/scylla-cdc-java/scylla-cdc-replicator/src/main/java/com/scylladb/cdc/replicator/Main.java:[78,17] sun.misc.Signal is internal proprietary API and may be removed in a future release
Warning:  /home/runner/work/scylla-cdc-java/scylla-cdc-java/scylla-cdc-replicator/src/main/java/com/scylladb/cdc/replicator/Main.java: sun.misc.Signal is internal proprietary API and may be removed in a future release
[INFO] /home/runner/work/scylla-cdc-java/scylla-cdc-java/scylla-cdc-replicator/src/main/java/com/scylladb/cdc/replicator/operations/ExecutingPreparedStatementHandler.java: Some input files use or override a deprecated API.
[INFO] /home/runner/work/scylla-cdc-java/scylla-cdc-java/scylla-cdc-replicator/src/main/java/com/scylladb/cdc/replicator/operations/ExecutingPreparedStatementHandler.java: Recompile with -Xlint:deprecation for details.
[INFO] /home/runner/work/scylla-cdc-java/scylla-cdc-java/scylla-cdc-replicator/src/main/java/com/scylladb/cdc/replicator/operations/ExecutingPreparedStatementHandler.java: Some input files use unchecked or unsafe operations.
[INFO] /home/runner/work/scylla-cdc-java/scylla-cdc-java/scylla-cdc-replicator/src/main/java/com/scylladb/cdc/replicator/operations/ExecutingPreparedStatementHandler.java: Recompile with -Xlint:unchecked for details.
[INFO] 
[INFO] --- resources:3.3.1:testResources (default-testResources) @ scylla-cdc-replicator ---
[INFO] skip non existing resourceDirectory /home/runner/work/scylla-cdc-java/scylla-cdc-java/scylla-cdc-replicator/src/test/resources
[INFO] 
[INFO] --- compiler:3.13.0:testCompile (default-testCompile) @ scylla-cdc-replicator ---
[INFO] Recompiling the module because of changed dependency.
[INFO] Compiling 2 source files with javac [debug target 11] to target/test-classes
[INFO] 
[INFO] --- surefire:3.5.4:test (default-test) @ scylla-cdc-replicator ---
[INFO] Using auto detected provider org.apache.maven.surefire.junitplatform.JUnitPlatformProvider
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0
```

--- a/scylla-cdc-lib/src/test/java/com/scylladb/cdc/lib/AlterTableIT.java
+++ b/scylla-cdc-lib/src/test/java/com/scylladb/cdc/lib/AlterTableIT.java
@@ -7,6 +7,7 @@ import com.datastax.driver.core.Cluster;
 import com.datastax.driver.core.PreparedStatement;
 import com.datastax.driver.core.Session;
 import com.google.common.base.Preconditions;
+import com.google.common.util.concurrent.Uninterruptibles;
 import com.scylladb.cdc.model.TableName;
 import com.scylladb.cdc.model.worker.ChangeSchema;
 import com.scylladb.cdc.model.worker.RawChange;
@@ -15,6 +16,7 @@ import java.net.InetSocketAddress;
 import java.util.List;
 import java.util.Properties;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.concurrent.atomic.AtomicReference;
@@ -50,25 +52,19 @@ public class AlterTableIT {
               keyspace, table));
 
       AtomicInteger counter = new AtomicInteger(0);
+      AtomicBoolean taskShouldStop = new AtomicBoolean(false);
       // Start continuously populating the base table in the background
       Runnable task =
           () -> {
-            while (!Thread.currentThread().isInterrupted()) {
-              PreparedStatement ps =
-                  session.prepare(
-                      String.format(
-                          "INSERT INTO %s.%s (column1, column2, column3) VALUES (?, ?, ?);",
-                          keyspace, table));
-              while (true) {
-                int current = counter.incrementAndGet();
-                session.execute(ps.bind(current, current, current));
-                try {
-                  // Makeshift rate limiting
-                  Thread.sleep(100);
-                } catch (InterruptedException e) {
-                  throw new RuntimeException(e);
-                }
-              }
+            PreparedStatement ps =
+                session.prepare(
+                    String.format(
+                        "INSERT INTO %s.%s (column1, column2, column3) VALUES (?, ?, ?);",
+                        keyspace, table));
+            while (!taskShouldStop.get()) {
+              int current = counter.incrementAndGet();
+              session.execute(ps.bind(current, current, current));
+              Uninterruptibles.sleepUninterruptibly(100, TimeUnit.MILLISECONDS);
             }
           };
       Thread thread = new Thread(task);
@@ -109,7 +105,12 @@ public class AlterTableIT {
       } catch (InterruptedException e) {
         e.printStackTrace();
       }
-      thread.interrupt();
+      taskShouldStop.set(true);
+      try {
+        thread.join();
+      } catch (InterruptedException e) {
+        throw new RuntimeException("Failed to join the background thread", e);
+      }
 
       // Verify the schema of first and last RawChange
       RawChange sample = firstChange.get();

